### PR TITLE
Task/aut 875/data testid attrs

### PR DESCRIPTION
**Pull request by AnastasiaArcadia**
**Ticket:** https://oat-sa.atlassian.net/browse/AUT-949

**How to test:**
- Create item (test, asset) class
- Add properties
- Check that properties controls (inputs, textareas, etc.) have data-testid attrs
- Check that e2e tests run successfully

_Export button was not affected as it's associated with changes in  the ui libs (like [here](https://github.com/oat-sa/tao-core-ui-fe/blob/master/src/loadingButton/tpl/button.tpl))_

**Deployed on:** http://test-anastasia-2.playground.kitchen.it.taocloud.org:45401

### CODE REVIEW

File: helpers/form/elements/xhtml/class.Button.php [hunk 1]
```diff
@@ -97,4 +97,10 @@ public function setType($type)
             $this->type = $type;
         }
     }
+
+    // Is the game worth the candle?
```
Comment (bartlomiejmarszal) [inline helpers/form/elements/xhtml/class.Button.php:101]: Yes it is. 
```suggestion
```

**Review by bartlomiejmarszal — APPROVED** [review]
BE changes approved

- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Review by Silvia-loza — COMMENTED** [review]
Confirmed that have data-testid attrs, however some of the tests don't pass when I run them locally,
 Perhaps something to do with  the environment I am testing with?

**Review by Silvia-loza — APPROVED** [review]
ALL looks good and all new Tests pass. Great job!
- [ ] New code is covered by tests (if applicable)
- [X] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [X] New code is respecting code style rules
- [X] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [X] Feature is working correctly on my local machine (if applicable)
- [X] Acceptance criteria are respected
- [X] Pull request title and description are meaningful

**Review by mike-ionut-mihai-sandu-tao — CHANGES_REQUESTED** [review]
please change the target of your branch 
also please remove the debug information

File: helpers/form/elements/xhtml/class.Button.php [hunk 2]
```diff
@@ -97,4 +97,10 @@ public function setType($type)
...
+    public function setTestId($testId)
+    {
+        $this->setAttribute('data-testid', $testId);
```
Comment (mike-ionut-mihai-sandu-tao) [inline helpers/form/elements/xhtml/class.Button.php:103]: please apply formating

File: helpers/form/elements/xhtml/class.Readonly.php [hunk 3]
```diff
@@ -43,7 +43,7 @@ class tao_helpers_form_elements_xhtml_Readonly extends tao_helpers_form_elements
     public function render()
     {
         $returnValue = $this->renderLabel();
-        $returnValue .= "<input type='text' readonly='readonly' disabled='disabled' name='{$this->name}' id='{$this->name}' ";
+        $returnValue .= "<input type='text' readonly='readonly' disabled='disabled' name='{$this->name}' id='{$this->name}' data-testid='{$this->getDescription()}' ";
```
Comment (mike-ionut-mihai-sandu-tao) [inline helpers/form/elements/xhtml/class.Readonly.php:46]: are you sure that you can still access $this->getDescription() here?

Comment (AnastasiaArcadia) [reply helpers/form/elements/xhtml/class.Readonly.php:46]: For me works
@pnal?

**Comment by mike-ionut-mihai-sandu-tao** [comment]
please add inside the ticket where we can test this (as you did changes to prod code we need to validate that tao GUI is not breaking) so QA needs to check that before closing the ticket

**Comment by mike-ionut-mihai-sandu-tao** [comment]
please also fix the tests as your change was broken them 
see here : http://qa.kitchen.it.taocloud.org/blue/organizations/jenkins/ci-pipeline-cg-extension-pr%2Ftao-core/detail/PR-3067/5/pipeline

**Review by mike-ionut-mihai-sandu-tao — APPROVED** [review]
lgtm

**Comment by mike-ionut-mihai-sandu-tao** [comment]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

**Pull request merged**